src/components/Dialog.tsx

```diff
@@ -99,15 +117,23 @@ export default class Dialog extends Component<DialogProps, DialogState> {
   }
 
   async componentDidMount() {
-    const { nacelleSpaceId: id, nacelleSpaceToken: token } = this.props.sdk
-      .parameters.installation as AppInstallationParameters
+    const {
+      nacelleSpaceId: id,
+      nacelleSpaceToken: token,
+      nacelleEndpoint: endpoint,
+    } = this.props.sdk.parameters.installation as AppInstallationParameters
     const invocation = this.props.sdk.parameters.invocation as DialogState
     const { value } = invocation
 
     // Set data handlers
-    await this.setClient(id, token)
+    await this.setClient(id, token, endpoint)
+    await this.setW2Settings(id, token, endpoint)
     await this.setStorage(id)
 
+    if (endpoint.includes('storefront')) {
+      this.setState(() => ({ isW2: true }))
+    }
```

Comment:
Looks like we are setting the client and storing the settings, regardless if it's W1 or W2. Is there a better way to handle this? W2 compatibility is checked later, so it still works.